src/type_checker/checker.rs

```diff
@@ -294,28 +294,35 @@ impl<B: Backend> TypeChecker<B> {
                             .var_name
                             .expect("anonymous lhs access cannot be mutated")
                     }
-                    _ => panic!("bad expression assignment (TODO: replace with error)"),
+                    _ => Err(self.error(
+                        ErrorKind::UnexpectedError("bad expression assignment"),
+                        expr.span,
+                    ))?,
                 };
 
                 // check that the var exists locally
                 let lhs_info = typed_fn_env
-                    .get_type_info(&lhs_name)?
-                    .expect("variable not found (TODO: replace with error")
+                    .get_type_info(&lhs_name)
+                    .or_else(|_| {
+                        Err(self.error(ErrorKind::UnexpectedError("variable not found"), expr.span))
+                    })?
                     .clone();
 
                 // and is mutable
-                if !lhs_info.mutable {
-                    return Err(self.error(ErrorKind::AssignmentToImmutableVariable, expr.span));
+                if let Some(lhs_info) = lhs_info {
```

Comment:
wait why is lhs_info still an option here

Reply:
I figured out what went wrong here: I didn't handle the `Result` with `?` produced from `get_type_info()`